src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -0,0 +1,47 @@
+package org.broadinstitute.dsde.rawls.dataaccess.slick
+
+import java.sql.Timestamp
+import java.util.UUID
+
+import org.joda.time.DateTime
+
+/**
+ * Created by dvoet on 2/4/16.
+ */
+case class WorkspaceRecord(
+  namespace: String,
+  name: String,
+  id: UUID,
+  bucketName: String,
+  createdDate: Timestamp,
+  lastModified: Timestamp,
+  createdBy: String,
+  isLocked: Boolean,
+  workspaceEntityId: Option[Long]
+)
+
+// TODO access control
+
+trait WorkspaceComponent {
+  this: DriverComponent with EntityComponent =>
+
+  import driver.api._
+
+  class WorkspaceTable(tag: Tag) extends Table[WorkspaceRecord](tag, "WORKSPACE") {
+    def namespace = column[String]("namespace")
+    def name = column[String]("name")
+    def id = column[UUID]("id", O.PrimaryKey)
+    def bucketName = column[String]("bucket_name")
+    def createdDate = column[Timestamp]("created_date")
+    def lastModified = column[Timestamp]("last_modified")
+    def createdBy = column[String]("created_by")
+    def isLocked = column[Boolean]("is_locked")
+    def workspaceEntityId = column[Option[Long]]("workspace_entity_id")
+
+    def workspaceEntity = foreignKey("FK_WORKSPACE_ENTITY", workspaceEntityId, entityQuery)(_.id.?)
```

Comment:
Interested in feedback on this. In order to implement workspace attributes we can use a special "workspace" entity. This foreign key is the reference to that entity. It is optional so because entity has a foreign key back to this table and one of the 2 needs to come first.

This works nicely because then workspaces can be treated as entities in every way and the "specialness" of them is reflected in the Workspace table.